core/lib/src/request/request.rs

```diff
@@ -386,6 +386,67 @@ impl<'r> Request<'r> {
             })
     }
 
+    /// Returns what protocol for a connection was handled by a proxy by
+    /// inspecting [`proto_header`](crate::Config::proto_header) of the request
+    /// if such a header is configured, exists and contains a valid setting
+    /// ("http" or "https").
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rocket::http::{Header, ForwardedProtocol};
+    ///
+    /// # let client = rocket::local::blocking::Client::debug_with(vec![]).unwrap();
+    /// # let request = client.get("/");
+    /// assert_eq!(request.forwarded_proto(), None);
+    ///
+    /// // `proto_header` defaults to `None`. `x-forwarded-proto` is considered the de-facto standard
+    /// # let figment = rocket::Config::figment().merge(("proto_header", "x-forwarded-proto"));
+    /// # let client = rocket::local::blocking::Client::debug(rocket::custom(figment)).unwrap();
+    /// # let request = client.get("/");
+    /// let request = request.header(Header::new("x-forwarded-proto", "https"));
+    /// assert_eq!(request.forwarded_proto(), Some(ForwardedProtocol::Https));
+    /// ```
+    pub fn forwarded_proto(&self) -> Option<ForwardedProtocol> {
+        let proto_header = self.rocket().config.proto_header.as_ref()?.as_str();
+        self.headers()
+            .get_one(proto_header)
+            .and_then(|proto| {
+                proto.parse()
+                    .ok()
+                    .map(|p| {
+                        if let ForwardedProtocol::Unregistered(_) = p {
+                            warn_!("'{proto_header}' header is malformed: {proto}");
+                        }
+                        p
+                    })
+            })
```

Comment:
```suggestion
            .and_then(|proto| proto.parse().ok())
```

I don't think we need this warning, and we certainly shouldn't emit it every time we call this.

Reply:
This has been removed by now. It was copied from `ip_header` earlier.